### PR TITLE
[ros2] Add JointTrajectory message conversion

### PR DESCRIPTION
**Pull request by AndrejOrsula**
This PR adds conversion between [`trajectory_msgs/msg/JointTrajectory`](https://github.com/ros2/common_interfaces/blob/foxy/trajectory_msgs/msg/JointTrajectory.msg) and `ignition.msgs.JointTrajectory` (PR - https://github.com/ignitionrobotics/ign-msgs/pull/106).

Direction `ROS 2 -> Ignition Transport` is tested with [`ign_moveit2`](https://github.com/AndrejOrsula/ign_moveit2). However, I did NOT explicitly try `Ignition Transport -> ROS 2` (besides running the automated tests).

CI fails because it is not using fork of `ign-msgs` with `ignition.msgs.JointTrajectory`, hence it cannot be found. I am unsure about the best way to resolve this problem. Any help/suggestions are appreciated :slightly_smiling_face:.

Also, I am currently not using ROS 1 and therefore I unfortunately won't be looking into adding this conversion there.

### CODE REVIEW

File: ros_ign_bridge/CMakeLists.txt
Comment (chapulina) [inline ros_ign_bridge/CMakeLists.txt]: Nitpick: We try to keep alphabetical order everywhere so it's easier to find missing / duplicate entries.

File: ros_ign_bridge/test/test_utils.hpp
Comment (chapulina) [inline ros_ign_bridge/test/test_utils.hpp]: Style nit: even though this is const, we've been using lowercase for all variables.

There are more instances below.

**Review by chapulina — COMMENTED** [review]
I haven't tried running it, but I took a high-level look and it looks good to me. I only noticed the small style issues pointed out below.

>  CI fails because it is not using fork of ign-msgs with ignition.msgs.JointTrajectory

The `ros2` branch supports Citadel (`ign-msgs5`) and Dome (`ign-msgs6`). So before merging this PR, https://github.com/ignitionrobotics/ign-msgs/pull/106 needs to be:

* merged against `ign-msgs6`
* backported to `ign-msgs5`
* released into `ign-msgs 5.X`
* released into `ign-msgs 6.X`

From your end, you just need to iterate on that PR until it gets merged, then the maintainers can take care of the rest.

> Also, I am currently not using ROS 1 and therefore I unfortunately won't be looking into adding this conversion there.

No problem, we can ticket an issue to give it visibility in case someone else wants to pick it up.

File: ros_ign_bridge/CMakeLists.txt
Comment (AndrejOrsula) [reply ros_ign_bridge/CMakeLists.txt]: Forgot to check this after [merging upstream to the branch](https://github.com/ignitionrobotics/ros_ign/pull/121/commits/3062a3319c45486efaf2577edda206fbdfbdafaa) :neutral_face:

Fixed in https://github.com/ignitionrobotics/ros_ign/pull/121/commits/5382c577999510ec569f97e5827d5508fd3cb406

File: ros_ign_bridge/test/test_utils.hpp
Comment (AndrejOrsula) [reply ros_ign_bridge/test/test_utils.hpp]: Renamed in https://github.com/ignitionrobotics/ros_ign/pull/121/commits/e785dc4508c919d154cea353776f9412a7355250

**Review by chapulina — COMMENTED** [review]
Looking good to me! Can I just ask that you sign all the commits on the PR for the DCO checker?

Then we just need to wait for the `ign-msgs` release and check CI.

**Review by chapulina — APPROVED** [review]
The necessary `ign-msgs` version is upstream and CI is passing. Merging!

**Pull request merged**
